audio_modder.py

```diff
@@ -2643,6 +2828,49 @@ def init_workspace(self):
         self.workspace.configure(yscrollcommand=self.workspace_scroll_bar.set)
         self.render_workspace()
 
+    def init_archive_search_bar(self):
+        if self.lookup_store == None:
+            logger.critical("Audio archive database connection is None after bypassing all check.",
+                            stack_info=True)
+            return
+        all_archives = self.lookup_store.query_helldiver_audio_archive()
+        self.archive_search = ArchiveSearch("{1} || {0}", 
+                                            entries=all_archives,
+                                            on_select_cb=self.on_archive_search_bar_return,
+                                            master=self.top_bar, 
+                                            width=64)
+        categories = self.lookup_store.query_helldiver_audio_archive_category()
+        categories = [""] + categories
+        self.category_search = ttk.Combobox(self.top_bar,
+                                            state="readonly",
+                                            font=('Segoe UI', 10),
+                                            width=18, height=10,
+                                            values=categories) 
+        self.archive_search.pack(side="left", padx=4, pady=8)
+        self.category_search.pack(side="left", padx=4, pady=8)
+        self.category_search.bind("<<ComboboxSelected>>",
+                                  self.on_category_search_bar_select)
+
+    def on_archive_search_bar_return(self, value: str):
```

Comment:
- This callback will only run when an `Entry` is focused, and `Return` is entered, which will select the current highlighted / selected option in the `ListBox`. 
- The risk of invalid input format reaching to this point is unlikely but a guard rail is placed behind it. 
  - As `Return` is received, the highlight / selected input is immediately passed into this callback unless there's some timing issue with the main event loop. 
  - If a user enter an input value that fully matches an option in the `ListBox` character by character, it will not be recognized because input value is used as fuzzy finding the option the users want to select, not as the raw input value for this callback.